File: v2/php/get_extraction.php
<?php

/**
 * Simple .env file loader.
 */
function loadEnv($dir) {
    $path = $dir . '/.env';
    if (!file_exists($path)) return;
    $lines = file($path, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    foreach ($lines as $line) {
        if (strpos(trim($line), '#') === 0) continue;
        if (strpos($line, '=') !== false) {
            list($name, $value) = explode('=', $line, 2);
            putenv(trim($name) . '=' . trim($value));
        }
    }
}

loadEnv(__DIR__);

// Parameter validation
if ($argc < 2) {
    die("Usage: php get_extraction.php <jobId>\n");
}

$jobId = $argv[1];

$apiKey = getenv('API_KEY');
$baseUrl = getenv('BASE_URL_API');

// Validate environment
if (!$apiKey || !$baseUrl) {
    die("Error: API_KEY or BASE_URL_API not set in environment.\n");
}

$url = $baseUrl . '/extractions/' . $jobId;

$ch = curl_init($url);

// Configure cURL for GET request
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    'Authorization: Bearer ' . $apiKey // API v2 uses Bearer Token
]);

$response = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

if (curl_errno($ch)) {
    echo 'Error: ' . curl_error($ch) . "\n";
} else {
    if ($httpCode >= 200 && $httpCode < 300) {
        echo "Extraction retrieved successfully!\n";
    } else {
        echo "Error: HTTP $httpCode\n";
    }
    
    // Output response
    $decoded = json_decode($response);
    echo json_encode($decoded, JSON_PRETTY_PRINT) . "\n";
}

curl_close($ch);

File: v2/python/get_extraction.py
import os
import json
import sys
import requests

def load_env(file_path=".env"):
    """
    Simple .env file loader that exports key=value pairs to os.environ.
    """
    if not os.path.exists(file_path):
        return
    with open(file_path, 'r') as f:
        for line in f:
            if '=' in line and not line.startswith('#'):
                key, value = line.strip().split('=', 1)
                os.environ[key] = value

def get_extraction(job_id):
    """
    Retrieve the current status and results of an extraction job.
    """
    load_env()
    api_key = os.getenv("API_KEY")
    base_url = os.getenv("BASE_URL_API")

    # Validate environment variables
    if not api_key or not base_url:
        print("Error: API_KEY and BASE_URL_API must be defined.")
        sys.exit(1)

    # API v2 uses Bearer Token authentication
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json"
    }

    print(f"Retrieving extraction for job {job_id}...")
    url = f"{base_url}/extractions/{job_id}"
    
    # Perform the GET request
    response = requests.get(url, headers=headers)

    print("\n--- Response ---")
    print(f"Status Code: {response.status_code}")
    try:
        print(json.dumps(response.json(), indent=2))
    except ValueError:
        print(response.text)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python get_extraction.py <job_id>")
        sys.exit(1)
    
    get_extraction(sys.argv[1])
